### PR TITLE
Cleanup pass through workflow files.

### PR DESCRIPTION
**Pull request by ScottTodd**
* Format files
* Pin dependencies (runner images, actions hashes)
* Reorder job steps (start with `actions/checkout`)
* Delete `test_shark_studio.yml` (this has been disabled for some time and it does not use any iree-turbine code)
* Add copyright/license header comments

### CODE REVIEW

**Review by marbre — APPROVED** [review]
Thanks for the cleanup.

File: .github/workflows/test_build_release.yml
Comment (ScottTodd) [inline .github/workflows/test_build_release.yml:1]: Failures in this `Build Release / Test Build Release Process (3.11, ubuntu-24.04)` workflow:

* Failed on two runs on this PR: https://github.com/iree-org/iree-turbine/actions/runs/12185371055/job/33991838631?pr=316
* Passing on `main`: https://github.com/iree-org/iree-turbine/actions/runs/12185526853/job/33991977314

Looking into it...

Comment (ScottTodd) [reply .github/workflows/test_build_release.yml:1]: Findings:

Before | After
-- | --
`ubuntu-latest` matched `ubuntu-22.04` | `ubuntu-24.04` explicitly
`setup-python@v3` installed `3.11.10` | `setup-python@v5.3.0` installed `3.11.10`

Before:
```
+ pip freeze
execnet==2.1.1
filelock==3.16.1
fsspec==2024.10.0
iniconfig==2.0.0
iree-base-compiler==3.1.0rc20241127
iree-base-runtime==3.1.0rc20241127
iree-turbine==2.5.0.dev0
Jinja2==3.1.4
MarkupSafe==3.0.2
ml_dtypes==0.5.0
mpmath==1.3.0
networkx==3.4.2
numpy==2.1.3
packaging==24.2
parameterized==0.9.0
pillow==11.0.0
pluggy==1.5.0
pytest==8.0.0
pytest-xdist==3.5.0
sympy==1.13.1
torch==2.5.1+cpu
torchvision==0.20.1+cpu
typing_extensions==4.12.2
```

After:
```
+ pip freeze
execnet==2.1.1
filelock==3.16.1
fsspec==2024.10.0
iniconfig==2.0.0
iree-base-compiler==3.1.0rc20241127
iree-base-runtime==3.1.0rc20241127
iree-turbine==2.5.0.dev0
Jinja2==3.1.4
MarkupSafe==3.0.2
ml_dtypes==0.5.0
mpmath==1.3.0
networkx==3.4.2
numpy==2.1.3
packaging==24.2
parameterized==0.9.0
pillow==11.0.0
pluggy==1.5.0
pytest==8.0.0
pytest-xdist==3.5.0
sympy==1.13.1
torch==2.5.1+cpu
torchvision==0.20.1+cpu
typing_extensions==4.12.2
```

Same packages, same Python version.

Comment (ScottTodd) [reply .github/workflows/test_build_release.yml:1]: Passed on 22.04.

The errors are suspicious...
```
>       buffer = hal_device.queue_alloca(alloc_size, wait_semaphores, signal_semaphores)
E       RuntimeError: allocating memory on queue: ABORTED

wheelhouse/test.venv/lib/python3.11/site-packages/iree/turbine/dynamo/tensor.py:307: RuntimeError
```

Another workflow is also failing: https://github.com/iree-org/iree-turbine/actions/runs/12185657821/job/33992398487?pr=316

I'll file an issue and switch back to 22.04 for now.

**Comment by Hardcode84** [comment]
Why do we want to pin GH actions hashes instead of exact versions like `v1.2.3`?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -21,48 +27,47 @@ jobs:
       fail-fast: false
       matrix:
         version: [3.11]
-        os: [ubuntu-latest]
+        os: [ubuntu-22.04]
```
Comment (ScottTodd) [inline .github/workflows/ci.yaml:30]: Ugh I'm going to need to change the required check for this...

image
image

**Comment by ScottTodd** [comment]
> Why do we want to pin GH actions hashes instead of exact versions like `v1.2.3`?

Rationale is at https://opensource.google/documentation/reference/github/services#actions

Tags can be moved easily while commit hashes can't. Pinning to a hash makes it so someone (e.g. a malicious actor) can't change the version that we use (e.g. to run malicious code). If someone _does_ attack an action like that, we'd be insulated and could hopefully see complaints from other users before we would be affected.

**Comment by ScottTodd** [comment]
More rationale for pinning actions versions: https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

**Pull request merged**